### PR TITLE
EVG-1995: agents should cancel stats collector threads when task collection is complete

### PR DESCRIPTION
**Pull request by tychoish**

### CODE REVIEW

File: agent/agent.go [hunk 1]
```diff
@@ -422,7 +423,6 @@ func (agt *Agent) Setup() error {
 	agt.KillChan = make(chan bool)
 	agt.metricsCollector = &metricsCollector{
 		comm: agt.TaskCommunicator,
-		stop: agt.KillChan,
```
Comment (bsamek) [inline agent/agent.go]: Does something still try to close this channel? I'm missing where that is.

File: config.go [hunk 2]
```diff
@@ -286,6 +286,11 @@ func NewSettings(filename string) (*Settings, error) {
 	if err != nil {
 		return nil, err
 	}
+
+	if err := settings.Validate(); err != nil {
```
Comment (bsamek) [inline config.go]: This is causing some of our tests to fail.

Comment (bsamek) [inline config.go]: Actually, I'm not sure. @john-m-liu Is your build in staging failing as a result of this line?

Comment (john-m-liu) [inline config.go]: yep, a lot of integration tests are not getting past the validation

[2017/08/17 16:21:44.975] --- FAIL: TestCLIFetchSource (0.00s)
[2017/08/17 16:21:44.977] panic: Error opening settings override file /data/mci/83081e45cbd076b6dba17762435b192e/gopath/src/github.com/evergreen-ci/evergreen/creds.yml: DBUrl and DB must not be empty [recovered]
[2017/08/17 16:21:44.977] 	panic: Error opening settings override file /data/mci/83081e45cbd076b6dba17762435b192e/gopath/src/github.com/evergreen-ci/evergreen/creds.yml: DBUrl and DB must not be empty

File: agent/agent.go [hunk 3]
```diff
@@ -606,6 +606,11 @@ func (agt *Agent) RunTask() (*apimodels.EndTaskResponse, error) {
 		grip.Debugf("stopping simple stats collector for task:", agt.GetCurrentTaskId())
 	}()
 
+	ctx, cancel := context.WithCancel(context.TODO())
+	defer cancel()
```
Comment (tychoish) [inline agent/agent.go]: We want to cancel before we finish and wait, but if the agent panics, (or we make a change that has us return early from this function) then we want to make sure that we cancel, but ideally we'll cancel on 643.

File: config.go (hunk 2 above)
Comment (tychoish) [inline config.go]: what garbage.

**Pull request merged**